src/main/java/com/pusher/client/connection/websocket/WebSocketClientWrapper.java

```diff
@@ -51,27 +52,27 @@ public WebSocketClientWrapper(final URI uri, final WebSocketListener proxy) thro
                 throw new SSLException(e);
             }
         }
-
-        this.proxy = proxy;
+        this.webSocketListener = webSocketListener;
+        setProxy(proxy);
```

Comment:
I'd definitely advocate calling regardless - always calling in to a third party library is easier to reason about that sometimes calling in to it. The fewer branches the better, especially hidden this low down.